src/main/java/org/isf/therapy/manager/TherapyManager.java

```diff
@@ -188,12 +187,10 @@ public TherapyRow newTherapy(TherapyRow thRow) throws OHServiceException {
 	@Transactional(rollbackFor=OHServiceException.class)
 	public boolean newTherapies(ArrayList<TherapyRow> thRows) throws OHServiceException {
 		if (!thRows.isEmpty()) {
-			DateTime now = new DateTime();
-
 
 			int patID = thRows.get(0).getPatient().getCode();
-			// FIXME FIXME			ioOperations.deleteAllTherapies(patID);
-			smsOp.deleteByModuleModuleID("therapy", String.valueOf(patID));
+			Patient patient = patientManager.getPatientById(patID);
+			ioOperations.deleteAllTherapies(patient);
```

Comment:
Only the comment was wrong, the operation was correct! 😅

Reply:
OK; let us try it one more time.